src/main/java/lavalink/client/io/jda/JdaLavalink.java

```diff
@@ -68,20 +68,32 @@ public JdaLink getExistingLink(Guild guild) {
     }
 
     /**
-     * May produce NullPointerException if JDA provider has not been initialised
+     * Returns the JDA instance with the {@code shardId shard ID}
+     *
+     * @param shardId the ID of the shard
+     * @return the JDA instance with the specified shard ID
+     *
+     * @throws IllegalStateException if the JDA provider has not been initialised
      */
     @SuppressWarnings({"WeakerAccess", "unused"})
-    @Nullable
+    @NonNull
     public JDA getJda(int shardId) {
+        if (jdaProvider == null) throw new IllegalStateException("JDA Provider not initialised!");
         return jdaProvider.apply(shardId);
     }
 
     /**
-     * May produce NullPointerException if JDA provider has not been initialised
+     * Returns the JDA instance with the {@code snowflake snowflake}
+     *
+     * @param snowflake the snowflake of the shard
+     * @return the JDA instance with the specified snowflake
+     *
+     * @throws IllegalStateException if the JDA provider has not been initialised
      */
     @SuppressWarnings("WeakerAccess")
-    @Nullable
+    @NonNull
     public JDA getJdaFromSnowflake(String snowflake) {
+        if (jdaProvider == null) throw new IllegalStateException("JDA Provider not initialised!");
         return jdaProvider.apply(LavalinkUtil.getShardFromSnowflake(snowflake, numShards));
     }
 
```

Comment:
Could be shortened to `return getJda(LavalinkUtil.getShardFromSnowflake(snowflake, numShards))`

Reply:
Yeah good point, I'll fix that in next commit